doc/releases/release-notes-3.5.rst

```diff
@@ -367,26 +367,164 @@ Drivers and Sensors
 Networking
 **********
 
-* Time and timestamps in the network subsystem, PTP and IEEE 802.15.4
-  were more precisely specified and all in-tree call sites updated accordingly.
-  Fields for timed TX and TX/RX timestamps have been consolidated. See
-  :c:type:`net_time_t`, :c:struct:`net_ptp_time`, :c:struct:`ieee802154_config`,
-  :c:struct:`ieee802154_radio_api` and :c:struct:`net_pkt` for extensive
-  documentation. As this is largely an internal API, existing applications will
-  most probably continue to work unchanged.
-
 * CoAP:
 
+  * Optimized CoAP client library to use only a single thread internally.
+  * Converted CoAP client library to use ``zsock_*`` API internally.
+  * Fixed a bug in CoAP client library, which resulted in an incorrect
+    retransmission timeout calculation.
   * Use 64 bit timer values for calculating transmission timeouts. This fixes potential problems for
     devices that stay on for more than 49 days when the 32 bit uptime counter might roll over and
     cause CoAP packets to not timeout at all on this event.
+  * API documentation improvements.
+  * Added new API functions:
+
+    * :c:func:`coap_has_descriptive_block_option`
+    * :c:func:`coap_remove_descriptive_block_option`
+    * :c:func:`coap_packet_remove_option`
+    * :c:func:`coap_packet_set_path`
+
+* Connection Manager:
+
+  * Added support for auto-connect and auto-down behaviors (controlled by
+    :c:enum:`CONN_MGR_IF_NO_AUTO_CONNECT` and :c:enum:`CONN_MGR_IF_NO_AUTO_DOWN`
+    flags).
+  * Split Connection Manager APIs into separate header files.
+  * Extended Connection Manager documentation to cover new functionalities.
+
+* DHCP:
+
+  * Added support for DHCPv4 unicast replies processing.
+  * Added support for DHCPv6 protocol.
+
+* ICMP:
+
+  * Fixed ICMPv6 error message type check.
+  * Reworked ICMP callback registration and handling, which allows to register
+    multiple handlers for the same ICMP message.
+  * Introduced an API to send ICMP Echo Request (ping).
+  * Added possibility to register offloaded ICMP ping handlers.
+  * Added support for setting packet priority for ping.
+
+* IPv6:
+
+  * Made sure that ongoing DAD procedure is cancelled when IPv6 address is removed.
+  * Fixed a bug, where Solicited-Node multicast address could be removed while
+    still in use.
 
 * LwM2M:
```

Comment:
btw I would add a mention to the new `lwm2m create` command - it's a great tool and worth advertising :)

Reply:
True, I've mentally linked it to IoP testing and skipped a separate entry for it, fixed now.